compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -397,28 +425,49 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
             self.path,
             AlignmentCheckFailed { required, has } =>
                 {
-                    "an unaligned {kind} (required {} byte alignment but found {})",
-                    required.bytes(),
-                    has.bytes(),
+                    found = "unaligned_ptr",
```

Comment:
How does this work, where does `unaligned_ptr` turn into something pretty?

More comments/docs seem needed here.

Reply:
No more stringly typed messages. Variants are now split to make it easy to translate and read.